sdk/core/core/src/az_json_parser.c

```diff
@@ -133,7 +133,22 @@ typedef struct
 
 AZ_NODISCARD static double az_json_number_to_double(az_dec_number const* p)
 {
-  return p->value * pow(10, p->exp) * p->sign;
+  double calculated_pow = 1;
+  if (p->exp > 0)
+  {
+    for (int i = 0; i < p->exp; i++)
+    {
+      calculated_pow *= 10;
+    }
+  }
+  else
+  {
+    for (int i = 0; i < -p->exp; i++)
+    {
+      calculated_pow /= 10;
+    }
+  }
+  return p->value * calculated_pow * p->sign;
```

Comment:
We lost the overflow check by switching away from pow().

Reply:
So, we can at most hold 2^64 into a uint64_t right..
witch is around 10^20 > x > 10^19
Would it be enough to add a validation about exp must be less than 20 or we throw OVERFLOW

Reply:
That won't catch everything.  
you should check that the previous value is less than the pow value before adding the sign.
(p->value * calculated_pow) > p->value